src/lua/init_path.rs

```diff
@@ -18,58 +18,64 @@ fn read_file<P: AsRef<Path>>(path: P) -> IOResult<File> {
 }
 
 #[cfg(test)]
-pub fn get_config() -> (bool, Result<File, &'static str>) {
+pub fn get_config() -> (bool, Result<(PathBuf, File), &'static str>) {
     (false, Err("Loading config should be ignored during tests for now"))
 }
 
 #[cfg(not(test))]
-pub fn get_config() -> (bool, Result<File, &'static str>) {
+pub fn get_config() -> (bool, Result<(PathBuf, File), &'static str>) {
     (true, get_config_file())
 }
 
 /// Parses environment variables
-fn get_config_file() -> Result<File, &'static str> {
+fn get_config_file() -> Result<(PathBuf, File), &'static str> {
     let home_var = env::var("HOME").expect("HOME environment variable not defined!");
     let home = home_var.as_str();
 
     if let Ok(path_env) = env::var("WAY_COOLER_INIT_FILE") {
         info!("Found $WAY_COOLER_INIT_FILE to be defined, will look for the init file there.");
-        if let Ok(file) = read_file(Path::new(&path_env)) {
+        let path = Path::new(&path_env);
+        if let Ok(file) = read_file(&path) {
             info!("Reading init file from $WAY_COOLER_INIT_FILE: {}",
                   path_env.as_str().replace(home, "~"));
-            return Ok(file)
+            // If the parent doesn't exist it's just in the current directory.
+            let dir = path.parent().map_or(PathBuf::new(), Path::to_path_buf);
```

Comment:
This changes how the `WAY_COOLER_INIT_FILE` env works, which I'm fine with but if you could note it in your main comment that would be nice. When I cut releases I usually just look at the root comment, and it would be nice to not have to dig through the code to document this there.